core/src/main/java/org/commonjava/maven/ext/core/impl/PreparseGroovyManipulator.java

```diff
@@ -0,0 +1,134 @@
+ * is stored in a {@link GroovyState} instance, which is in turn stored in the {@link ManipulationSession}.
+ */
+@Named("groovy-injection-preparse")
+@Singleton
+public class PreparseGroovyManipulator
+                extends BaseGroovyManipulator
+    //implements Manipulator
+{
+    @Inject
+    public PreparseGroovyManipulator( ModelIO modelIO, FileIO fileIO, PomIO pomIO )
+    {
+        super( modelIO, fileIO, pomIO );
+    }
+
+    /**
+     * Initialize the {@link GroovyState} state holder in the {@link ManipulationSession}. This state holder detects
+     * version-change configuration from the Maven user properties (-D properties from the CLI) and makes it available
+     * for later.
+     */
+    //@Override
+    public void init( final ManipulationSession session ) throws ManipulationException
+    {
+        final State state = new GroovyState( session.getUserProperties() );
+        this.session = session;
+        session.setState( state );
+
+        try
+        {
+            final GalleyInfrastructure infra = new GalleyInfrastructure( session, new DefaultMirrorSelector() );
+            infra.init( null, null, Files.createTempDirectory( "pme-" ).resolve( "cache-dir" ).toFile() );
+            final GalleyAPIWrapper wrapper = new GalleyAPIWrapper( infra );
+            modelIO = new ModelIO( wrapper );
+            fileIO = new FileIO( infra );
+        }
+        catch ( IOException e )
+        {
+            throw new ManipulationException( e.getMessage(), e );
+        }
+    }
+
+    /**
+     * Apply the groovy script changes.
+     */
+    //@Override
+    public Set<Project> applyChanges( final List<Project> projects )
+        throws ManipulationException
+    {
+        final GroovyState state = session.getState( GroovyState.class );
+
+        if ( !session.isEnabled() || !state.isEnabled() )
+        {
+            logger.debug("{}: Nothing to do!", getClass().getSimpleName() );
+            return Collections.emptySet();
+        }
+
+        final List<File> groovyScripts = parseGroovyScripts( state.getGroovyScripts() );
+
+        for ( final File groovyScript : groovyScripts )
+        {
+            final GroovyShell shell = new GroovyShell();
+
+            try
+            {
+                final Script script = shell.parse(groovyScript);
+                final InvocationPoint invocationPoint = script.getClass().getAnnotation(InvocationPoint.class);
+                final InvocationStage stage = invocationPoint != null ? invocationPoint.invocationPoint() : null;
+
+                if ( InvocationStage.PREPARSE == stage )
+                {
+                    applyGroovyScript( Collections.emptyList(), null, groovyScript );
+                }
+            }
+            catch ( IOException e )
+            {
+                throw new ManipulationException( e.getMessage(), e );
+            }
```

Comment:
I think both of these above catch clauses could be removed if just `applyGroovyScripts` is used?

Reply:
Done.